### ISSUE TITLE
[no-unused-vars] is not correct in d.ts files

### ISSUE THREAD

**Issue by Mister-Hope** [typescript-eslint/typescript-eslint#2300]
**Repro**

```yml
extends:
  - eslint:recommended

overrides:
  - files:
      - "*.ts"

    extends:
      - plugin:@typescript-eslint/eslint-recommended
      - plugin:@typescript-eslint/recommended
      - plugin:@typescript-eslint/recommended-requiring-type-checking

    plugins:
      - "@typescript-eslint"

    parserOptions:
      parser: "@typescript-eslint/parser"
      project:
        - "./tsconfig.json"
        - "./test-tsconfig.json"
```

shims-tsx.d.ts:

```TS
import Vue, { VNode } from "vue";

declare global {
  namespace JSX {
    type Element = VNode;
    type ElementClass = Vue;
    interface IntrinsicElements {
      // eslint-disable-next-line @typescript-eslint/no-explicit-any
      [elem: string]: any;
    }
  }
}
```

**Expected Result**

**Actual Result**

```
warning: 'Vue' is defined but never used (@typescript-eslint/no-unused-vars) at src\typings\shims-tsx.d.ts:1:8:
> 1 | import Vue, { VNode } from "vue";
    |        ^
  2 |
  3 | declare global {
  4 |   namespace JSX {


warning: 'VNode' is defined but never used (@typescript-eslint/no-unused-vars) at src\typings\shims-tsx.d.ts:1:15:
> 1 | import Vue, { VNode } from "vue";
    |               ^
  2 |
  3 | declare global {
  4 |   namespace JSX {
```

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `3.6.1` |
| `@typescript-eslint/parser`        | `3.6.1` |
| `TypeScript`                       | `3.9.6` |
| `ESLint`                           | `7.4.0` |
| `node`                             | `12.18.2` |
| `npm`                              |  :x: |
| `yarn`                              |  `1.22.4` |

**Comment by bradzacher** [comment]
Please use the search before filing new issues
#2292, #2026, #2287, #1678, #1596, #1302, etc.

https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+%22no-unused-vars%22+global+

Also please read the pinned issues before filing new issues - #1856
